fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/AvroDimensionRowParser.java

```diff
@@ -0,0 +1,185 @@
+import java.util.HashSet;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Parses an AVRO file into Dimension Rows.
+ */
+public class AvroDimensionRowParser {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AvroDimensionRowParser.class);
+    private static ObjectMapper MAPPER = null;
+
+    private DimensionFieldMapper dimensionFieldMapper;
+    private Map<String, String> convertedDimensionFields;
+
+    /**
+     * Constructs an AvroDimensionRowParser object based on the DimensionFieldMapper object.
+     *
+     * @param dimensionFieldMapper Object that defines the dimension field name transformations
+     * @param objectMapper Object that is used to construct mapper objects
+     */
+    public AvroDimensionRowParser(DimensionFieldMapper dimensionFieldMapper, ObjectMapper objectMapper) {
+        this.dimensionFieldMapper = dimensionFieldMapper;
+        MAPPER = objectMapper;
+    }
+
+    /**
+     * Validates the schema of the given AVRO file with the Dimension schema configured by the user.
+     *
+     * @param dimension The dimension object used to configure the dimension
+     * @param avroSchemaPath The path of the AVRO schema file (.avsc)
+     *
+     * @return true if the schema is valid, false otherwise
+     *
+     * @throws NullPointerException thrown if JSON object `fields` is not present
+     * @throws IOException thrown if there is error parsing the avro files
+     */
+    public boolean isSchemaValid(Dimension dimension, String avroSchemaPath) throws NullPointerException, IOException {
+
+        // Convert the AVRO schema file into JsonNode Object
+        JsonNode jsonAvroSchema = convertFileToJsonNode(avroSchemaPath);
+
+        // If the JSON schema file is corrupted/ empty, then return the schema is invalid
+        if (jsonAvroSchema == null) {
+            return false;
+        }
+
+        // This may throw a NullPointerException, if the AVRO schema file does not have a `fields` JSON object.
+        int fieldSize = jsonAvroSchema.get("fields").size();
+
+        jsonAvroSchema = jsonAvroSchema.get("fields");
+
+        HashSet<String> avroFields = new HashSet<>(fieldSize);
+
+        // Streaming cannot be used, because `avroFields` is a JsonNode object
+        for (int i = 0 ; i < fieldSize; i++) {
+            // Parsing the JsonNode object as Text, `toString()` adds an additional `"`
+            avroFields.add(jsonAvroSchema.get(i).get("name").asText());
+        }
+
+        Function<DimensionField, String> curriedConverter = curry(dimension, dimensionFieldMapper);
+
+        // Generates a map of configured dimension field names and the converted field names, for easier lookups in the
+        // `parseAvroFileDimensionRows()`
+        convertedDimensionFields = dimension.getDimensionFields()
+                .stream()
+                .collect(Collectors.toMap(dimensionField -> dimensionField.getName(), curriedConverter));
+
```

Comment:
This is a major side effect that your downstream code depends on happening inside a validate method.  That is some major spooky-action-at-a-distance.  Either build the convertedDimensionFields explicitly, pass it as a variable, or throw away the copy you build for this validator and rebuild the convertedFields closer to where they'll be used later.